packages/core/src/storage.ts

```diff
@@ -19,16 +19,22 @@ import type { Tx } from './tx'
 // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
 export type QuerySelector<T> = {
   $in?: T[]
+  $like?: string
 }
+
 export type ObjQueryType<T> = T | QuerySelector<T>
 
-export type Query<T> = {
+export type DocumentQuery<T extends Doc> = {
   [P in keyof T]?: ObjQueryType<T[P]>
 }
 
-export type DocumentQuery<T extends Doc> = Query<T>
-
 export interface Storage {
   findAll: <T extends Doc>(_class: Ref<Class<T>>, query: DocumentQuery<T>) => Promise<T[]>
   tx: (tx: Tx) => Promise<void>
 }
+
+export function likeToRegExp (value: string): RegExp {
```

Comment:
@BykhovDenis Let's not leak abstraction, we need a function to check if value is matches like.
So le't do it, match only. We do not need to return regexp, since If you would like to change implementation to custom one like implementation, you will also need to rewrite all places where it is used.